types/funcs.go

```diff
@@ -27,3 +27,43 @@ type FuncContext func(context.Context)
 
 // FuncContextError takes a context and may return an error
 type FuncContextError func(context.Context) error
+
+// IsFuncType tests if a function is one of FuncType
+func IsFuncType(fn interface{}) bool {
+	switch fn.(type) {
+	case func():
+		return true
+	case func() error:
+		return true
+	case func(context.Context):
+		return true
+	case func(context.Context) error:
+		return true
+	}
+	return false
+}
+
+// FuncTypeWrap wraps a valid FuncType to FuncContextError
+func FuncTypeWrap(fn interface{}) FuncContextError {
```

Comment:
so I don't think the named type FuncContextError really pays for itself here. It doesn't have any special meaning.  I think it's clearer just to make this 
```
func FuncTypeWrap(fn interface{}) func(context.Context) error {
```

Then you don't have to look up what FuncContextError is to figure out what this is returning.  And since the returned function is not special in any way, it's nice not fool people into thinking it's something special by giving it a name.